### PR TITLE
Polkadot 9.30.0

### PR DESCRIPTION
**Pull request by nitro-marky**
This replaces the previous version we were using (0.9.5). Deprecations have been replaced and tests have been updated so they now work.

The biggest thing to notice is things like Event, Origin, Call etc mainly begin with Runtime now.

It also looks like updating to 9.3.0 means installing protoc in the build pipeline

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -17,57 +17,68 @@ targets = ['x86_64-unknown-linux-gnu']
 [dependencies]
 clap = { version = "3.1.18", features = ["derive"] }
 bs58 = "0.4.0"
+protoc-rust = "2.28.0"
```
Comment (mattdean-digicatapult) [inline node/Cargo.toml:20]: Is this necessary? I know it was added to a bunch of other places in error

File: pallets/transaction-payment-free/src/lib.rs [hunk 2]
```diff
@@ -124,4 +125,48 @@ where
...
+    }
+
+    fn metadata() -> Vec<sp_runtime::traits::SignedExtensionMetadata> {
+        sp_std::vec![SignedExtensionMetadata {
+            identifier: Self::IDENTIFIER,
+            ty: scale_info::meta_type::<Self>(),
+            additional_signed: scale_info::meta_type::<Self::AdditionalSigned>()
+        }]
+    }
 }
```
Comment (mattdean-digicatapult) [inline pallets/transaction-payment-free/src/lib.rs:172]: Any idea where these changes have come from?

File: runtime/Cargo.toml [hunk 3]
```diff
@@ -62,17 +62,21 @@ pallet-symmetric-key = { version = '2.0.0', default-features = false, package =
...
 [build-dependencies]
-substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.25" }
+substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.30" }
 
 [features]
 default = ["std"]
 std = [
+    "frame-try-runtime?/std",
+	"frame-system-benchmarking?/std",
+	"frame-benchmarking?/std",
```
Comment (mattdean-digicatapult) [inline runtime/Cargo.toml:72]: `Cargo.toml` files need formatting. Sorry I think this instance is my bad

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -454,8 +459,8 @@ construct_runtime!(
         Preimage: pallet_preimage,
         Scheduler: pallet_scheduler,
         IpfsKey: pallet_symmetric_key,
-        Membership: pallet_membership::<Instance1>,
-        TechnicalCommittee: pallet_collective::<Instance1>,
+        TechnicalCommittee: pallet_collective<Instance1>,
+        Membership: pallet_membership<Instance1>,
```
Comment (mattdean-digicatapult) [inline runtime/src/lib.rs:463]: needs reverting!

File: node/Cargo.toml (hunk 1 above)
Comment (nitro-marky) [reply node/Cargo.toml]: I think this can go now.

File: pallets/transaction-payment-free/src/lib.rs (hunk 2 above)
Comment (nitro-marky) [reply pallets/transaction-payment-free/src/lib.rs:172]: These were copied over from the template as I thought they would be needed, but I can get rid if not the case.

File: node/src/command_helper.rs [hunk 5]
```diff
@@ -22,7 +22,7 @@
 use crate::service::FullClient;
```
Comment (mattdean-digicatapult) [inline node/src/command_helper.rs:22]: Can we move to `benchmarking.rs` to match the node template please?

File: runtime/src/lib.rs [hunk 6]
```diff
@@ -210,8 +212,7 @@ impl pallet_aura::Config for Runtime {
 }
 
 impl pallet_grandpa::Config for Runtime {
-    type Event = Event;
-    type Call = Call;
+    type RuntimeEvent = RuntimeEvent;
```
Comment (dblane-digicatapult) [inline runtime/src/lib.rs:215]: Should there not be a `type RuntimeCall` ?

File: pallets/doas/src/lib.rs [hunk 7]
```diff
@@ -63,7 +60,7 @@ pub mod pallet {
         /// # </weight>
         #[pallet::weight({
           let dispatch_info = call.get_dispatch_info();
-          (dispatch_info.weight.saturating_add(10_000), dispatch_info.class)
```
Comment (mattdean-digicatapult) [inline pallets/doas/src/lib.rs:66]: why was this removed, can you remember?

File: pallets/doas/src/lib.rs [hunk 8]
```diff
@@ -115,7 +112,6 @@ pub mod pallet {
           let dispatch_info = call.get_dispatch_info();
           (
             dispatch_info.weight
-              .saturating_add(10_000)
```
Comment (mattdean-digicatapult) [inline pallets/doas/src/lib.rs:118]: same here, can you remember why this changed?

File: pallets/transaction-payment-free/src/lib.rs [hunk 9]
```diff
@@ -1,8 +1,9 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 use codec::{Decode, Encode};
-use frame_support::weights::{DispatchInfo, PostDispatchInfo};
+use frame_support::dispatch::{DispatchInfo, PostDispatchInfo};
 use scale_info::TypeInfo;
+use sp_runtime::traits::SignedExtensionMetadata;
```
Comment (mattdean-digicatapult) [inline pallets/transaction-payment-free/src/lib.rs:6]: linting is sayng `sp_runtime::traits::SignedExtensionMetadata` is unused. Try removing?

File: .github/workflows/release-types-package.yml [hunk 10]
```diff
@@ -87,7 +87,7 @@ jobs:
         run: npm run depcheck
 
   check-version:
-    name: 'Check version'
+    name: "Check version"
```
Comment (jonmattgray) [inline .github/workflows/release-types-package.yml:90]: there's some formatting in the workflow that looks good, removing white space and fixing indents. I'm not sure about the change from single `'` to double `"`. The convention seems to be single `'` except for in `run:` commands

**Review by mattdean-digicatapult — COMMENTED** [review]
Round 2, this time I've been a bit more thorough. Note most of these are genuine questions!

File: node/src/command_helper.rs (hunk 5 above)
Comment (nitro-marky) [reply node/src/command_helper.rs:22]: 👍

Comment (mattdean-digicatapult) [reply node/src/command_helper.rs:22]: as in rename the whole file

File: pallets/transaction-payment-free/src/lib.rs (hunk 9 above)
Comment (nitro-marky) [reply pallets/transaction-payment-free/src/lib.rs:6]: Yeah, just caught that.

File: pallets/doas/src/lib.rs (hunk 8 above)
Comment (nitro-marky) [reply pallets/doas/src/lib.rs:118]: I should have checked but I was getting these errors and wasn't sure how to sort it
`sp_weights::weight_v2::Weight
pub const fn saturating_add(self, rhs: Self) -> Self

Saturating [Weight](vscode-file://vscode-app/Users/MAddis/Documents/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) addition. Computes self + rhs, saturating at the numeric bounds of all fields instead of overflowing.
Go to [Weight](vscode-file://vscode-app/Users/MAddis/Documents/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)
error: mismatched types
label: arguments to this function are incorrect
note: associated function defined here
label: arguments to this function are incorrect`

File: pallets/doas/src/lib.rs (hunk 7 above)
Comment (nitro-marky) [reply pallets/doas/src/lib.rs:66]: Back in now.

File: pallets/doas/src/lib.rs (hunk 8 above)
Comment (nitro-marky) [reply pallets/doas/src/lib.rs:118]: Back in now

File: .github/workflows/release-types-package.yml (hunk 10 above)
Comment (nitro-marky) [reply .github/workflows/release-types-package.yml]: I've left the cargo.toml files as we normally have them

File: runtime/src/lib.rs (hunk 6 above)
Comment (dblane-digicatapult) [reply runtime/src/lib.rs:215]: yes but we are removing a call the pallets runtime config and not replacing it?  This doesn't feel correct.

File: .github/workflows/release-types-package.yml (hunk 10 above)
Comment (nitro-marky) [reply .github/workflows/release-types-package.yml]: It only seems to affect Node.

File: node/src/command.rs [hunk 11]
```diff
@@ -12,7 +12,7 @@ use sp_keyring::Sr25519Keyring;
 
 impl SubstrateCli for Cli {
     fn impl_name() -> String {
-        "DSCP Node".into()
+        "DSCP Node ".into()
```
Comment (mattdean-digicatapult) [inline node/src/command.rs:15]: why the extra space?

File: pallets/doas/src/lib.rs [hunk 12]
```diff
@@ -59,14 +59,20 @@ pub mod pallet {
         /// - Weight of derivative `call` execution + 10,000.
         /// # </weight>
         #[pallet::weight({
-          let dispatch_info = call.get_dispatch_info();
+          let dispatch_info = Runtimecall.get_dispatch_info();
           (dispatch_info.weight, dispatch_info.class)
         })]
-        pub fn doas_root(origin: OriginFor<T>, call: Box<<T as Config>::Call>) -> DispatchResultWithPostInfo {
+        pub fn doas_root(origin: OriginFor<T>, Runtimecall: Box<<T as Config>::Call>) -> DispatchResultWithPostInfo {
```
Comment (mattdean-digicatapult) [inline pallets/doas/src/lib.rs:65]: this one should be `call` not `Runtimecall`. And the casing is wrong anyway

File: pallets/doas/src/lib.rs [hunk 13]
```diff
@@ -59,14 +59,20 @@ pub mod pallet {
...
+            let dispatch_info = Runtimecall.get_dispatch_info();
+            (
+                dispatch_info.weight.saturating_add(Weight::from_ref_time(10_000)),
+                dispatch_info.class
+            );
+
```
Comment (mattdean-digicatapult) [inline pallets/doas/src/lib.rs:71]: where has this come from?

File: node/src/command.rs (hunk 11 above)
Comment (nitro-marky) [reply node/src/command.rs:15]: `Error:   --> /home/runner/work/dscp-node/dscp-node/node/src/command.rs:15:15
   |
15 |         'DSCP Node'.into()
   |               ^^^^ unknown prefix
   |
   = note: prefixed identifiers and literals are reserved since Rust 2021
help: consider inserting whitespace here
   |
15 |         'DSCP Node '.into()
   |                   +
`

Comment (mattdean-digicatapult) [reply node/src/command.rs:15]: that's because of single-quotes not because of the space. Line should be
```rust
"DSCP Node".into()
```
which builds fine for me locally

File: pallets/doas/src/lib.rs (hunk 13 above)
Comment (mattdean-digicatapult) [reply pallets/doas/src/lib.rs]: I'll look into this I think

**Review by mattdean-digicatapult — DISMISSED** [review]

File: api/package.json [hunk 14]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@digicatapult/dscp-node",
-  "version": "4.3.1",
+  "version": "4.3.2",
```
Comment (mattdean-digicatapult) [inline api/package.json:3]: `4.3.2` -> `4.4.0`

**Review by mattdean-digicatapult — APPROVED** [review]
